src/main/java/org/apache/mesos/hdfs/state/LiveState.java

```diff
@@ -71,25 +73,23 @@ public void updateTaskForStatus(Protos.TaskStatus status) {
       // If initializing the first NN or reconciling the first NN or bootstrapping the first NN
       // set the status to initialized
       if (status.getMessage().equals(HDFSConstants.NAME_NODE_INIT_MESSAGE)
-          || (currentAcquisitionPhase.equals(AcquisitionPhase.RECONCILING_TASKS)
-              && !isNameNode1Initialized())
-          || (status.getMessage().equals(HDFSConstants.NAME_NODE_BOOTSTRAP_MESSAGE)
-              && !isNameNode1Initialized())) {
+          || (currentAcquisitionPhase.equals(AcquisitionPhase.RECONCILING_TASKS) && !isNameNode1Initialized())          //NOPMD  indicates useless parens
+          || (status.getMessage().equals(HDFSConstants.NAME_NODE_BOOTSTRAP_MESSAGE) && !isNameNode1Initialized())) {    //NOPMD
+
         nameNode1TaskMap.clear();
         nameNode1TaskMap.put(status, true);
-      } // If bootstrapping the second NN or reconciling the second NN,
+
+        // If bootstrapping the second NN or reconciling the second NN,
         // set the status to initialized
-      else if ((status.getMessage().equals(HDFSConstants.NAME_NODE_BOOTSTRAP_MESSAGE)
-          && !isNameNode2Initialized())
-          || (currentAcquisitionPhase.equals(AcquisitionPhase.RECONCILING_TASKS)
-              && !isNameNode2Initialized())) {
+      } else if ((status.getMessage().equals(HDFSConstants.NAME_NODE_BOOTSTRAP_MESSAGE) && !isNameNode2Initialized())   //NOPMD
```

Comment:
It looks really weird to me to see the comments for the following `else if` statement inside the previous `if` statement. Can we move them down, inside the else if which it describes?